ethicml/metrics/accuracy.py

```diff
@@ -41,3 +41,25 @@ class F1(SklearnMetric):
     sklearn_metric = (f1_score,)
     apply_per_sensitive: ClassVar[bool] = True
     _name: ClassVar[str] = "F1"
+
+
+@dataclass
+class RobustAccuracy(SklearnMetric):
+    """Classification accuracy."""
+
+    sklearn_metric = (accuracy_score,)
+    apply_per_sensitive: ClassVar[bool] = False
+    _name: ClassVar[str] = "Robust Accuracy"
+
+    @implements(SklearnMetric)
+    def score(self, prediction: Prediction, actual: DataTuple) -> float:
+        scores = []
+        for _s in actual.s[actual.s.columns[0]].unique():
+            preds_subset = Prediction(hard=prediction.hard[actual.s[actual.s.columns[0]] == _s])
+            data_subset = DataTuple(
+                x=actual.x[actual.s[actual.s.columns[0]] == _s],
+                s=actual.s[actual.s[actual.s.columns[0]] == _s],
+                y=actual.y[actual.s[actual.s.columns[0]] == _s],
+            )
```

Comment:
Hm, I feel we must have a method for applying a mask to a data tuple

Reply:
yeah, I'd have though so, we do it in enough places